tests/model_serving/model_server/maas_billing/utils.py

```diff
@@ -249,20 +262,207 @@ def verify_chat_completions(
+            "rates": [{"limit": 50, "window": "2m"}],
+            "when": [{"predicate": 'auth.identity.tier == "enterprise"'}],
+        },
+        "free": {
+            "counters": [{"expression": "auth.identity.userid"}],
+            "rates": [{"limit": 5, "window": "1m"}],
+            "when": [{"predicate": 'auth.identity.tier == "free"'}],
+        },
+        "premium": {
+            "counters": [{"expression": "auth.identity.userid"}],
+            "rates": [{"limit": 8, "window": "1m"}],
+            "when": [{"predicate": 'auth.identity.tier == "premium"'}],
+        },
+    }
+
 
-    assert isinstance(content_text, str) and content_text.strip(), (
-        "First choice in /v1/chat/completions response has no text content"
+@contextmanager
+def maas_gateway_rate_limits_patched(
+    *,
+    admin_client: DynamicClient,
+    namespace: str,
+    token_policy_name: str,
+    request_policy_name: str,
+) -> Generator[None, None, None]:
+    """
+    Temporarily patch ONLY 'spec.limits' of the Kuadrant
+    TokenRateLimitPolicy and RateLimitPolicy for MaaS tests,
+    and restore the original state afterwards.
+    """
+    token_policy = TokenRateLimitPolicy(
+        client=admin_client,
+        name=token_policy_name,
+        namespace=namespace,
     )
+    request_policy = RateLimitPolicy(
+        client=admin_client,
+        name=request_policy_name,
+        namespace=namespace,
+    )
+
+    LOGGER.info(f"MaaS Kuadrant: using policies {namespace}/{token_policy_name} and {namespace}/{request_policy_name}")
+
+    token_body = token_policy.instance.to_dict()
+    request_body = request_policy.instance.to_dict()
+
+    if "resourceVersion" in token_body["metadata"]:
+        del token_body["metadata"]["resourceVersion"]
+    if "resourceVersion" in request_body["metadata"]:
+        del request_body["metadata"]["resourceVersion"]
+
+    token_body.setdefault("spec", {})["limits"] = maas_token_ratelimitpolicy_spec()
+    request_body.setdefault("spec", {})["limits"] = maas_ratelimitpolicy_spec()
+
+    patches = {
+        token_policy: token_body,
+        request_policy: request_body,
+    }
+
+    LOGGER.info(f"Patching Kuadrant policies in namespace '{namespace}' via ResourceEditor")
+
+    with ResourceEditor(patches=patches):
+
+        def _wait_for_enforced(policy):
+            for condition in policy.instance.status.conditions:
+                if condition.type in ("Enforced", "Ready") and condition.status == "True":
+                    return True
+            return False
+
+        def _check_enforced():
+            return _wait_for_enforced(token_policy) and _wait_for_enforced(request_policy)
+
+        LOGGER.info("Waiting for patched policies to be Enforced/Ready...")
+        sampler = TimeoutSampler(
+            wait_timeout=60,
+            sleep=5,
+            func=_check_enforced,
+        )
+        for sample in sampler:
+            if sample:
+                break
+
+        yield
```

Comment:
https://github.com/RedHatQE/openshift-python-wrapper/blob/main/ocp_resources/resource.py#L1242 this is the one I asked you to use during our call last week. Please use it.